src/libraries/System.Security.Cryptography.Algorithms/tests/ChaCha20Poly1305Tests.cs

```diff
@@ -0,0 +1,464 @@
+                    "6c206f7220746f206369746520746865" +
+                    "6d206f74686572207468616e20617320" +
+                    "2fe2809c776f726b20696e2070726f67" +
+                    "726573732e2fe2809d").HexToByteArray(),
+                AssociatedData = "f33388860000000000004e91".HexToByteArray(),
+                Ciphertext = (
+                    "64a0861575861af460f062c79be643bd" +
+                    "5e805cfd345cf389f108670ac76c8cb2" +
+                    "4c6cfc18755d43eea09ee94e382d26b0" +
+                    "bdb7b73c321b0100d4f03b7f355894cf" +
+                    "332f830e710b97ce98c8a84abd0b9481" +
+                    "14ad176e008d33bd60f982b1ff37c855" +
+                    "9797a06ef4f0ef61c186324e2b350638" +
+                    "3606907b6a7c02b0f9f6157b53c867e4" +
+                    "b9166c767b804d46a59b5216cde7a4e9" +
+                    "9040c5a40433225ee282a1b0a06c523e" +
+                    "af4534d7f83fa1155b0047718cbc546a" +
+                    "0d072b04b3564eea1b422273f548271a" +
+                    "0bb2316053fa76991955ebd63159434e" +
+                    "cebb4e466dae5a1073a6727627097a10" +
+                    "49e617d91d361094fa68f0ff77987130" +
+                    "305beaba2eda04df997b714d6c6f2c29" +
+                    "a6ad5cb4022b02709b").HexToByteArray(),
+                Tag = "eead9d67890cbb22392336fea1851f38".HexToByteArray()
+            }
+        };
+    }
+
+    public class ChaCha20Poly1305IsSupportedTests
+    {
+        public static bool RuntimeSaysIsNotSupported => !ChaCha20Poly1305.IsSupported;
+
+        [ConditionalFact(nameof(RuntimeSaysIsNotSupported))]
+        public static void CtorThrowsPNSEIfNotSupported()
+        {
+            byte[] key = RandomNumberGenerator.GetBytes(256 / 8);
+
+            Assert.Throws<PlatformNotSupportedException>(() => new ChaCha20Poly1305(key));
+            Assert.Throws<PlatformNotSupportedException>(() => new ChaCha20Poly1305(key.AsSpan()));
+        }
+
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Browser)]
+        public static void IsSupported_OnBrowser_ReturnsFalse()
+        {
+            Assert.False(ChaCha20Poly1305.IsSupported);
+        }
+
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public static void IsSupported_OnWindows_QueriesOSSupport()
+        {
+            // Runtime uses a hardcoded OS version to determine support.
+            // The test queries the OS directly to ensure our version check is correct.
+
+            bool expected = CngUtility.IsAlgorithmSupported("CHACHA20_POLY1305");
+            Assert.Equal(expected, ChaCha20Poly1305.IsSupported);
+        }
+
+        [Fact]
+        [SkipOnPlatform(TestPlatforms.Browser | TestPlatforms.Windows, "We special-case Browser and Windows checks.")]
+        public static void IsSupported_OnOtherPlatforms_ReturnsFalse()
+        {
+            Assert.False(ChaCha20Poly1305.IsSupported);
+        }
+    }
```

Comment:
As with above, it'd be nice to squish the IsSupported into one test so there's not a positive and a negative list of the same thing.

```C#
[Fact]
public static void CheckIsSupported()
{
    bool shouldWork = false;

    if (Environment.IsWindows())
    {
        shouldWork = CngUtility.IsAlgorithmSupported("CHACHA20_POLY1305");
    }

    if (shouldWork)
    {
        Assert.True(ChaCha20Poly1305.IsSupported, "ChaCha20Poly1305.IsSupported");
    }
    else
    {
        Assert.False(ChaCha20Poly1305.IsSupported, "ChaCha20Poly1305.IsSupported");
    }
}
```

Then when the Unix/OpenSSL version is added, and this test fails, all that has to be done is update the initial false to an environment expression (or there's another complex grey area test)

}